src/makeRobot.js

```diff
@@ -37,8 +37,65 @@
  *
  * @return {Robot}
  */
+
 function makeRobot(name, wheels, version) {
-  // write code here
+  const robot = {
+    name,
+    wheels,
+    version,
+    coords: {
+      x: 0,
+      y: 0,
+    },
+    get info() {
+      return 'name: ' + this.name
+        + ', chip version: ' + this.version
+        + ', wheels: ' + this.wheels;
+    },
+    get location() {
+      return `${this.name}: x=${this.coords.x}, y=${this.coords.y}`;
+    },
+    goForward(distance = 1) {
+      if (distance < 0) {
+        return this;
+      }
+      this.coords.y += distance;
```

Comment:
it can be simplified
```suggestion
      this.coords.y += distance < 0 ? 0 : distance;
```